src/CheckOut.js

```diff
@@ -161,17 +184,47 @@ class CheckOut extends React.Component {
     this.state = { loading: false, blocked: false };
     this.patronFormRef = React.createRef();
     this.timer = undefined;
+    this.shouldSubmitAutomatically = hasIn(location, 'state.patronBarcode') && hasIn(location, 'state.itemBarcode');
     this.state = { submitting: false };
   }
 
   componentDidMount() {
-    this.patronFormRef.current.focus();
+    const {
+      setFieldValue,
+      location,
+    } = this.props;
+
+    if (this.shouldSubmitAutomatically) {
+      setFieldValue('patronForm', 'patron.identifier', location.state.patronBarcode);
+      setFieldValue('itemForm', 'item.barcode', location.state.itemBarcode);
+    } else {
+      this.patronFormRef.current.focus();
+    }
   }
 
   componentDidUpdate(prevProps) {
-    const patronBlocks = get(this.props.resources, ['patronBlocks', 'records'], []);
-    const prevBlocks = get(prevProps.resources, ['patronBlocks', 'records'], []);
+    const {
+      submitForm,
+      patronBarcodeFieldValue,
+      mutator,
+      resources,
+    } = this.props;
+
+    const {
+      resources: prevResources,
+      patronBarcodeFieldValue: prevPatronBarcodeFieldValue,
+    } = prevProps;
+
     const { submitting } = this.state;
+
+    const patronFormFilled = !prevPatronBarcodeFieldValue && patronBarcodeFieldValue;
+
+    if (patronFormFilled && this.shouldSubmitAutomatically) {
+      submitForm('patronForm');
+    }
+
+    const patronBlocks = get(resources, ['patronBlocks', 'records'], []);
+    const prevBlocks = get(prevResources, ['patronBlocks', 'records'], []);
```

Comment:
Does this cause automatic-submit to happen before checking for patron-blocks? Without a parallel PR in ui-checkin that provides the location-state, I couldn't test that scenario. I asked the same question [on the ticket](https://issues.folio.org/browse/UICHKIN-109).